js/ProductCat.js

```diff
@@ -258,38 +268,20 @@
     applyPrototypesCall($productcat.q[i]);
   }
 
-  /**
-   * Get query parameters from url [URI] object
-   */
-  function getQueryParamValue(uri, name) {
-    return find([`__vi_${name}`, name], () => uri.getQueryParamValue(name));
-  }
 
   // Set the status to indicate loaded success
   $productcat.loaded = true;
-
+  // export in browser environment
   if (typeof window !== 'undefined') {
-    // // For browser env
-    // // Check the link of current page, to check if the page is come from a previous search click.
-    // let refReqId = null;
-    // if (document.referrer) {
-    //   const refUri = new URI(document.referrer);
-    //   refReqId = getQueryParamValue(refUri, QUERY_REQID);
-    // }
-    // // Get reqid from current url or from its referrer url
-    // const curUri = new URI(window.location.href);
-    // const reqid = getQueryParamValue(curUri, QUERY_REQID) || refReqId;
-    // const imName = getQueryParamValue(curUri, QUERY_IMNAME);
-    // // If reqid, action and imName are found in current page content, send event tracking.
-    // if (reqid && imName) {
-    //   // Get event 'action' for current page, will set it as 'click' action by default if not specified.
-    //   const action = getQueryParamValue(curUri, QUERY_ACTION) || 'click';
-    //   // sendEvent(EVENTS.actions, {
-    //   //   reqid: getQueryParamValue(curUri, QUERY_REQID),
-    //   //   action,
-    //   //   im_name: getQueryParamValue(curUri, QUERY_IMNAME),
-    //   // });
-    // }
+    // get uid from local storage.
+    if (typeof(Storage) !== 'undefined') {
+      let uid = localStorage.getItem("uid")
+      if (!uid) { // if not exist in local storage generate and save in localstorage.
+        uid = generateUUID();
+        localStorage.setItem("uid", uid);
+      }
+      $productcat.set("uid", uid);
```

Comment:
this will store into settings ? correct ?

Reply:
need to allow user to override our uid setting also. Is this supported? Only if they do not provide the uid, we will generate random UUID.

Reply:
yes, so in browser environment, it will check if uid is set in localstorage, if not set, set it and in store in settings. 
else, retrieve uid from previous set value.

Reply:
yes but this code is ignoring the user settings e.g. I want to pass my own UID based on login username for example, in Android, they can call setUid() method. Does not seem possible here.